### ISSUE TITLE
No dimming support in google home actions for matter device 

### ISSUE THREAD

**Issue by smhc** [arendst/Tasmota#21958]
### PROBLEM DESCRIPTION

A "Light 1 Dimmer" (or Light 3 RGB) matter device cannot have the brightness adjusted via Google Home actions/scripts. The brightness can be adjusted through the device "tile" in Google Home directly, but the brightness cannot be adjusted via a routine. When using a script the validation suggests the device does not support "BrightnessAbsoluteCommand" or "BrightnessRelativeCommand" commands, only the OnOff command.

### REQUESTED INFORMATION

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): esp32-12f
- [x] Tasmota binary firmware version number used: esp32c3
  - [x] Pre-compiled
  - [ ] Self-compiled
- [x] Flashing tools used: web
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:

10:26:36.568 MQT: stat/tasmota_58E99C/RESULT = {"NAME":"rgbwlight","GPIO":[1,1,32,418,417,1,1,1,1,1,1,0,0,0,0,0,0,0,416,419,1,1],"FLAG":0,"BASE":1}
10:26:36.805 MQT: stat/tasmota_58E99C/RESULT = {"Module":{"0":"rgbwlight"}}
10:26:37.009 MQT: stat/tasmota_58E99C/RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"32":"Button1"},"GPIO3":{"418":"PWM3"},"GPIO4":{"417":"PWM2"},"GPIO5":{"0":"None"},"GPIO6":{"0":"None"},"GPIO7":{"0":"None"},"GPIO8":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO11":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO18":{"416":"PWM1"},"GPIO19":{"419":"PWM4"},"GPIO20":{"0":"None"},"GPIO21":{"0":"None"}}

```
- [x] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:

no rules

```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:

10:28:24.623 MQT: stat/tasmota_58E99C/STATUS = {"Status":{"Module":0,"DeviceName":"CurtainLight","FriendlyName":["CurtainLight",""],"Topic":"tasmota_58E99C","ButtonTopic":"0","Power":"01","PowerLock":"00","PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
10:28:24.631 MQT: stat/tasmota_58E99C/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota32/release/tasmota32c3.bin","RestartReason":"Software reset digital core","Uptime":"0T02:03:32","StartupUTC":"2024-08-14T07:24:52","Sleep":50,"CfgHolder":4617,"BootCount":13,"BCResetTime":"2024-08-14T07:37:03","SaveCount":141}}
10:28:24.639 MQT: stat/tasmota_58E99C/STATUS2 = {"StatusFWR":{"Version":"14.1.0.4(102428f-tasmota32)","BuildDateTime":"2024-08-13T13:17:54","Core":"3_0_4","SDK":"5.1.4.240801","CpuFrequency":160,"Hardware":"ESP32-C3 v0.3","CR":"403/699"}}
10:28:24.646 MQT: stat/tasmota_58E99C/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["FARMW",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001800600003C5A0A192800000000","00000080","00006000","00004000","00000020"]}}
10:28:24.660 MQT: stat/tasmota_58E99C/STATUS4 = {"StatusMEM":{"ProgramSize":1985,"Free":894,"Heap":141,"StackLowMark":2,"PsrMax":0,"PsrFree":0,"ProgramFlashSize":4096,"FlashSize":4096,"FlashChipId":"164020","FlashFrequency":80,"FlashMode":"DIO","Features":["0809","9F9AD7DF","0015A001","B7F7BFCF","05DA9BC4","E0360DC7","480840D2","20200000","D4BC482D","810A80B1","00000015"],"Drivers":"1,2,!3,4,!5,7,!8,9,10,11,12,!14,!16,!17,!20,!21,!24,26,!27,29,!34,!35,38,50,52,!59,!60,62,!63,!66,!67,!68,!71,!73,82,!86,!87,!88,!121","Sensors":"1,2,3,5,6,7,8,9,10,11,12,13,14,15,17,18,19,20,21,22,26,31,34,37,39,40,42,43,45,51,52,55,56,58,59,64,66,67,74,85,92,95,98,103,105,109,127","I2CDriver":"7,8,9,10,11,12,13,14,15,17,18,20,24,29,31,36,41,42,44,46,48,58,62,65,69,76,77,82"}}
10:28:24.675 MQT: stat/tasmota_58E99C/STATUS5 = {"StatusNET":{"Hostname":"tasmota-58E99C-2460","IPAddress":"192.168.1.38","Gateway":"192.168.1.1","Subnetmask":"255.255.255.0","DNSServer1":"8.8.8.8","DNSServer2":"8.8.4.4","Mac":"A0:76:4E:58:E9:9C","IP6Global":"fd06:8df3:e47f:940:a276:4eff:fe58:e99c","IP6Local":"fe80::a276:4eff:fe58:e99c%st1","Ethernet":{"Hostname":"","IPAddress":"0.0.0.0","Gateway":"0.0.0.0","Subnetmask":"0.0.0.0","DNSServer1":"8.8.8.8","DNSServer2":"8.8.4.4","Mac":"00:00:00:00:00:00","IP6Global":"","IP6Local":""},"Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":19.0}}
10:28:24.682 MQT: stat/tasmota_58E99C/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.1.118","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_58E99C","MqttUser":"DVES_USER","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
10:28:24.692 MQT: stat/tasmota_58E99C/STATUS7 = {"StatusTIM":{"UTC":"2024-08-14T09:28:24Z","Local":"2024-08-14T10:28:24","StartDST":"2024-03-31T02:00:00","EndDST":"2024-10-27T03:00:00","Timezone":"+01:00","Sunrise":"05:42","Sunset":"20:06"}}
10:28:24.701 MQT: stat/tasmota_58E99C/STATUS10 = {"StatusSNS":{"Time":"2024-08-14T10:28:24"}}
10:28:24.712 MQT: stat/tasmota_58E99C/STATUS11 = {"StatusSTS":{"Time":"2024-08-14T10:28:24","Uptime":"0T02:03:32","UptimeSec":7412,"Heap":141,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Berry":{"HeapUsed":23,"Objects":281},"POWER1":"ON","Dimmer1":77,"POWER2":"OFF","Dimmer2":49,"Color":"C4005B7D","HSBColor":"332,100,77","White":49,"Channel":[77,0,36,49],"Scheme":0,"Fade":"OFF","Speed":1,"LedTable":"ON","Wifi":{"AP":1,"SSId":"FARMW","BSSId":"D4:DA:21:5B:61:1A","Channel":11,"Mode":"HT20","RSSI":98,"Signal":-51,"LinkCount":1,"Downtime":"0T00:00:04"}}}

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

1. Install latest version of "Google Home" on Android 14 / pixel 7. It may be necessary to enter the preview programme - but I have seen the same behaviour either way. I am using a "Google Nest Hub 2nd Gen Smart Home Display" as the hub.
2. Create a new "home" in the app to contain the device. This seems to be important as old "homes" appear to still be compatible.
3. Create a "Light 1 Dimmer" device on Matter, real or virtual. (Note I have a RGBW device so have a Light 3 RGB and a Light 1 Dimmer).
4. Connect the device to Google Home app
5. Attempt to create a routine using either the wizard or scripting to adjust the brightness of the device.

### EXPECTED BEHAVIOUR

The ability to adjust the brightness of the light should be available to routines.

### SCREENSHOTS

![Screenshot_20240814-193104](https://github.com/user-attachments/assets/5a0d1770-c2d4-4eb8-b39e-200a4262b28d)

### ADDITIONAL CONTEXT
The device can still have its brightness adjusted via the "tile" in google home with the following appearing in the logs:

```
08:27:12.870 MQT: stat/tasmota_58E99C/COMMAND = {"MtrReceived":{"4":{"Ep":4,"Name":"MyLight","Bri":79,"Dimmer":31,"Power":1}}}
08:27:14.346 MQT: stat/tasmota_58E99C/COMMAND = {"MtrReceived":{"4":{"Ep":4,"Name":"MyLight","Bri":254,"Dimmer":100,"Power":1}}}
08:27:15.493 MQT: stat/tasmota_58E99C/COMMAND = {"MtrReceived":{"4":{"Ep":4,"Name":"MyLight","Bri":155,"Dimmer":61,"Power":1}}}
```

However the routines cannot adjust the device brightness. Other light devices in my Google home can have their brightness adjusted, although these are not matter devices. I have tested in both bridged and 'static' non-bridged mode with the same behaviour observed.
I am unable to test against other matter devices.

```
_matter_device.json

{"distinguish":3934,"passcode":37241527,"ipv4only":false,"disable_bridge_mode":true,"nextep":5,
"config":{"2":{"type":"light3"},"3":{"type":"light1","light":2},"4":{"name":"MyLight","type":"v_light1"}}}
```

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by smhc** [comment]
I noticed this other issue: https://github.com/arendst/Tasmota/issues/21947
if discovery is broken there's a good chance it is related to this problem.

**Comment by s-hadinger** [comment]
I doubt it's related to #21947 
I cannot test for the next two weeks. It looks like it may be a bug on Google side, considering dimmer control does work through the tile

**Comment by smhc** [comment]
Ok thanks, enjoy your time off. I tried repairing the devices using Tasmota 13.4.0 and still couldn't adjust the brightness. Would be good to try a different matter light device but I don't have anything on hand. I might try to create a virtual one with node red.

**Comment by smhc** [comment]
I finally got mvd working and confirmed brightness adjustment was possible with that virtual device. But when I tried the tasmota device again it was also working there. So it may have been a temporary server side issue. Or maybe something else kicked it into working correctly.. either way it seems to be working ok now.

![Screenshot](https://github.com/user-attachments/assets/98d60685-7c0a-4a15-9545-4557115d824e)
